test/src/test/java/org/corfudb/integration/StreamingPatternsIT.java

```diff
@@ -343,6 +343,89 @@ public void testDefaultPortionOfResumeOrDefaultPolicy() throws Exception {
         assertThat(shutdownCorfuServer(corfuServer)).isTrue();
     }
 
+    /**
+     * This test verifies the 'default' re-subscription functionality for the resumeOrDefault when
+     * resume is not possible after freeing the table data immediately upon subscription.
+     * It validates that the table metadata is not removed when freeTableData() is called.
+     */
+    @Test
+    public void testDefaultPortionOfResumeOrDefaultWithFreePolicy() throws Exception {
```

Comment:
Several comments regarding tests/verification:

1. This test is a copy/paste of other ITs with one added line, to avoid code dupe, isn't it better to generalize it and have a boolean that indicates if table freed or not, and that would cover both ITs. 

Also, seems much simpler to test this, to open a table, sync it.. free table and now subscribe the listener. It would isolate the test to what we really want to test cause this test does way more things testing the listener logic + checkpoint and so on...

2. I don't see a test that confirms that after using the freeTable API we verify it is rebuilding the table correctly on new access (I think this one is important).

Reply:
1. Sure will add a boolean and refactor the test
2. This test does exist in current PR please see `CorfuStoreShimTest::checkFreeTableData`